pytorch_lightning/core/lightning.py

```diff
@@ -855,11 +855,15 @@ def init_ddp_connection(self):
         try:
             root_node = os.environ['SLURM_NODELIST'].split(' ')[0]
         except Exception:
-            root_node = '127.0.0.2'
+            root_node = '127.0.0.1'
 
         root_node = self.trainer.resolve_root_node_address(root_node)
         os.environ['MASTER_ADDR'] = root_node
-        torch_distrib.init_process_group('nccl', rank=proc_rank, world_size=world_size)
+        if self.trainer.on_gpu:
```

Comment:
why the different backends?

Reply:
NCCL only works for GPU. The PyTorch docs recommend using GLOO for CPU training: https://pytorch.org/docs/stable/distributed.html#backends